### ISSUE TITLE
importRdata - designmatrix, comparisonstomake

### ISSUE THREAD

**Issue by sydneyguy1981** [kvittingseerup/IsoformSwitchAnalyzeR#93]
Hi. I'm having some issues (and confusion) in relation to the arguments for designmatrix and comparisonstomake.

I can import the kallisto data with:
`Txi_trans_F <- importIsoformExpression(sampleVector = path)`

I can assign column headers to abundance and counts with:
`colnames(Txi_trans_F$abundance) <- c("isoform_id", sampleLabels)`
`colnames(Txi_trans_F$counts) <- c("isoform_id", sampleLabels)`

I create my design:
`myDesign <- data.frame(
  sampleID = colnames(Txi_trans_F$abundance)[-1],
  condition = gsub('_.*', '', colnames(Txi_trans_F$abundance)[-1])
)`

which gives me:
sampleID                     condition
1 Control_Female_1      Control
2 Control_Female_2     Control
3 Control_Female_3     Control
4  Stress_Female_1       Stress
5  Stress_Female_2      Stress
6  Stress_Female_3      Stress

I then create comparisonstomake with:
`comparisonsToMake = as.data.frame(cbind(condition_1 = c("Stress"),
                                        condition_2 = c("Control")))`

which gives me:
  condition_1 condition_2
1      Stress     Control

But when I run importRdata using:
`mySwitchList_F <- importRdata(
  isoformCountMatrix   = Txi_trans_F$counts,
  isoformRepExpression = Txi_trans_F$abundance,
  designMatrix         = myDesign,
  removeNonConvensionalChr = TRUE,
  addAnnotatedORFs=TRUE,
  comparisonsToMake = comparisonsToMake,
  isoformExonAnnoation = "Mus_musculus.GRCm38.101.chr_patch_hapl_scaff.gtf.gz",
  isoformNtFasta       = "Mus_musculus.GRCm38.cdna.all.fa",
  showProgress = TRUE)`

the output is:
        comparison        estimated_genes_with_dtu
1 Control - Stress                  16 - 26

when it should be stress - control. Please advise.

Thanks!

**Comment by kvittingseerup** [comment]
That does indeed sound strange. Could you please post the result of running:

```head(mySwitchList_F, 2)```

Also are you sure you want the base-line to be "Stress". Does that not make the intrepretation of the direction of change hard to intrepret?

**Comment by sydneyguy1981** [comment]
Hi Kristoff. Thanks for the response and sincerest apologies for the double-post on the google groups as well. I have deleted my earlier post on this platform.

Here is the output from importRdata:
Step 1 of 7: Checking data...
Step 2 of 7: Obtaining annotation...
    importing GTF (this may take a while)
Done.
    24519 ( 20.84%) isoforms were removed since they were not expressed in any samples.
Step 3 of 7: Fixing StringTie gene annoation problems...
    There were no need to fix any annotation
Step 4 of 7: Calculating gene expression and isoform fractions...
Step 5 of 7: Merging gene and isoform expression...
Step 6 of 7: Making comparisons...
Step 7 of 7: Making switchAnalyzeRlist object...
The GUESSTIMATED number of genes with differential isoform usage are:
        comparison estimated_genes_with_dtu
1 Control - Stress                  16 - 26
Done

And here is the output from `head(mySwitchList_F,2`

iso_ref          gene_ref            isoform_id               gene_id condition_1
1 isoComp_00000001 geneComp_00000001  ENSMUST00000000001.4  ENSMUSG00000000001.4      Stress
2 isoComp_00000002 geneComp_00000002 ENSMUST00000000028.13 ENSMUSG00000000028.15      Stress
  condition_2 gene_name   gene_biotype    iso_biotype gene_overall_mean gene_value_1
1     Control     Gnai3 protein_coding protein_coding         19.373808    18.573532
2     Control     Cdc45 protein_coding protein_coding          5.185872     5.443493
  gene_value_2 gene_stderr_1 gene_stderr_2 gene_log2_fold_change gene_q_value iso_overall_mean
1    20.174083     0.6223625     1.4836101             0.1191933           NA       19.3738076
2     4.928251     2.2540207     0.5246095            -0.1431806           NA        0.8527387
  iso_value_1 iso_value_2 iso_stderr_1 iso_stderr_2 iso_log2_fold_change iso_q_value IF_overall
1  18.5735319   20.174083    0.6223625    1.4836101           0.11919335          NA  1.0000000
2   0.8691783    0.836299    0.1201351    0.2078959          -0.05498834          NA  0.1904667
        IF1       IF2     dIF isoform_switch_q_value gene_switch_q_value   PTC
1 1.0000000 1.0000000  0.0000                     NA                  NA FALSE
2 0.2025667 0.1783667 -0.0242                     NA                  NA FALSE

In response to your comment, for the contrast matrix i'm using for limma, i'm using:
`contrast.matrix_F <- makeContrasts(condition = Stress - Control,
                                 levels=design_F)`

I was under the assumption that i'd need to use the same type of contrast for subsequent DEXSeq analysis with isoformSwitchAnalysisCombined. Please advise if i'm incorrect.

Thanks and happy new year.

**Comment by kvittingseerup** [comment]
Ahh. But when you say `Stress - Control` with limma it is using "Control" as baseline.

The reason for this is that limma are doing the models in log space so log(X) - log(Y) and since this is the same as log(X/Y) you can see that Y is considered baseline meaning the fold changes are compared to this.

This is not the case for IsoformSwitchAnalyzeR - here condition 1 is used as baseline.

So while it is strange that IsoformSwitchAnalyzeR did not change it based on the `comparisonsToMake` argument your analysis is actually the right one.

Which version of IsoformSwitchAnalyzeR are you using?

For future references you can also format tabels or function output the using the Rmarkdown style you also you for the functions :)

**Comment by sydneyguy1981** [comment]
Hi Kristoff. Thank you for the prompt reply and the clarification.

Following your advice, I've changed the code to:

`comparisonsToMake = as.data.frame(cbind(condition_1 = c("Control"),
                                        condition_2 = c("Stress")),
                                        stringsAsFactors = F)`

From importRdata, I was able to obtain:
_comparison estimated_genes_with_dtu
1 Stress - Control                  47 - 79_

However, when I ran, isoformSwitchAnalysisCombined:
`mySwitchList_F <- isoformSwitchAnalysisCombined(
  switchAnalyzeRlist   = mySwitchListFiltered_F,
  switchTestMethod = 'DEXSeq',
  pathToOutput = 'isoform_output_F', # directory must already exist
  outputSequences = TRUE)`

all the outputs are Control vs Stress:
_The number of isoform switches found were:
         Comparison nrIsoforms nrSwitches nrGenes
1 Control vs Stress         84         88      70_

_The number of isoform switches with functional consequences identified were:
         Comparison nrIsoforms nrSwitches nrGenes
1 Control vs Stress         64         71      54_

Also, all the switchplots in the the 'isoform_output_F' folder were labelled Control vs Stress.

So is Control still baseline in these figures and data comparisons?



The version i'm using: IsoformSwitchAnalyzeR_1.12.0
